STYLEGUIDE.md

```diff
@@ -0,0 +1,215 @@
+# Particle Cloud Framework Style Guide
+
+This is a Python Style Guide for the PCF Framework. The intent of this guide is to strive for cleaner and better code quality. Following this guide, our code will be approachable to anyone who needs to maintain it later, including ourselves!
+
+## Table of Contents
+1. [Virtual Environments](#venv)
+2. [Imports](#imports)
+3. [Naming](#naming)  
+    i. [Variables](#variables)  
+    ii. [Class names](#classes)  
+    iii. [Functions/Methods](#funcs)
+4. [Spacing](#spacing)
+5. [Strings](#strings)  
+    i. [Quotes](#quotes)  
+    ii. [Docstrings](#docstr)
+6. [Dictionary Key Value Retrieval](#dict)
+7. [PCF Utility Functions](#pcf_util)
+8. [PCF Exceptions](#pcf_excep)
+
+## <a name="venv">Virtual Environments</a>
+------
+### Conda
+Conda is a package management system that can help you easily keep track of the package requirements 
+for projects and install them automatically for you. 
+You can set up and share collections of packages called environments.
+
+For sake of clarity, all references will be intended for MacOS users. 
+
+#### Installation
+Visit this <a href="https://docs.anaconda.com/anaconda/install/">link</a> to 
+find installation details for your appropriate OS.
+
+<a href="https://www.anaconda.com/download/#macos"> MacOs</a>  
+
+Click *version 3.7*
+
+### Create a new virtual env
+Visit this 
+<a href="https://conda.io/docs/user-guide/getting-started.html#starting-conda">link</a> 
+to find documentation on your appropriate OS.
+
+All conda commands will be typed in the Terminal window.
+
+```commandline
+conda create --name pcf-local python==3.6
+```
+
+*PCF requires Python version 3.6+*
+
+### Activate virutal env
+```commandline
+source activate pcf-local
+```
+
+### Deactivate virtual env
+```commandline
+source deactivate
+```
+
+### View list of exisiting virtual envs
+```commandline
+conda info --envs
+```
+
+## <a name="imports">Imports</a>
```

Comment:
Adding something about the order of imports might be good for readability. Most of what we are writing in this comes from PEP8, but it bears repeating.

https://www.python.org/dev/peps/pep-0008/#imports

Reply:
We do not fully follow PEP8 order of import standards. Our current unofficial practice is usually a mix. In terms of importing first and then applying relative imports we aim to follow the standard. However, the process of ordering imports based on the type of import (standard library, 3rd party, etc) is not required. This begs a different question. Why aren't we prioritizing this?